session/xmlrpc.go

```diff
@@ -82,8 +79,8 @@ func (x *XmlRpcTransport) DoRequest(
 		}
 
 		timeout := DefaultXmlRpcTimeout
-		if x.Timeout != 0 {
-			timeout = x.Timeout
+		if sess.Timeout != 0 {
```

Comment:
Not sure I like that it is essentially impossible to override timeout to be 0 (unlimited).